CMakeLists.txt

```diff
@@ -98,19 +104,32 @@ download_external_sources()
 
 
 # Googletest
-set(gtest_force_shared_crt ON CACHE BOOL "Use shared run-time")
-add_external_subdirectory(googletest)
-firebase_ios_add_alias(GTest::GTest gtest)
-firebase_ios_add_alias(GTest::Main gtest_main)
-firebase_ios_add_alias(GMock::GMock gmock)
+if(FIREBASE_IOS_BUILD_TESTS)
+  set(gtest_force_shared_crt ON CACHE BOOL "Use shared run-time")
+  add_external_subdirectory(googletest)
+  firebase_ios_add_alias(GTest::GTest gtest)
+  firebase_ios_add_alias(GTest::Main gtest_main)
+  firebase_ios_add_alias(GMock::GMock gmock)
+endif()
 
 
 # Benchmark
-set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Firestore disabled")
-set(BENCHMARK_ENABLE_EXCEPTIONS OFF CACHE BOOL "Firestore disabled")
-set(BENCHMARK_ENABLE_INSTALL OFF CACHE BOOL "Firestore disabled")
-set(BENCHMARK_ENABLE_GTEST_TESTS OFF CACHE BOOL "Firestore disabled")
-add_external_subdirectory(benchmark)
+if(FIREBASE_IOS_BUILD_BENCHMARKS)
+  set(BENCHMARK_ENABLE_TESTING OFF CACHE BOOL "Firestore disabled")
+  set(BENCHMARK_ENABLE_EXCEPTIONS OFF CACHE BOOL "Firestore disabled")
+  set(BENCHMARK_ENABLE_INSTALL OFF CACHE BOOL "Firestore disabled")
+  set(BENCHMARK_ENABLE_GTEST_TESTS OFF CACHE BOOL "Firestore disabled")
+
+  if(IOS)
```

Comment:
FYI, IOS is only a standard CMake variable as of version 3.14.  For the C++ repo, we had to explicitly define IOS via the toolchain file to get around this (https://github.com/firebase/firebase-cpp-sdk/blob/master/cmake/toolchains/ios.cmake#L44), though will probably work on changing that to use a more explicit TARGET_PLATFORM_IOS, or something similar.  You might want to do something similar as well, to support older cmakes.

Reply:
The "standard" ios toolchain file also sets `IOS`: https://github.com/leetal/ios-cmake/blob/master/ios.toolchain.cmake#L466

Coding to modern CMake and adding fallback support for older CMake versions seems like a reasonable path forward. If/when `firebase-cpp-sdk` chooses to implement some other convention, we can revisit here too.